### ISSUE TITLE
Class redefinition failure when loading new class.

### ISSUE THREAD

**Issue by Anon10W1z** [raphw/byte-buddy#41]
new ByteBuddy().redefine(BlockDirt.class).method(named("updateTick")).intercept(MethodDelegation.to(CppDirtInterceptor.class)).make().load(BlockDirt.class.getClassLoader(), ClassReloadingStrategy.fromInstalledAgent());

Upon calling the load method this crashes, despite the CppDirtInterceptor containing a method taking the exact parameters that the updateTick method in the class takes.

```
java.lang.UnsupportedOperationException: class redefinition failed: attempted to add a method
	at sun.instrument.InstrumentationImpl.redefineClasses0(Native Method) ~[?:1.8.0_40]
	at sun.instrument.InstrumentationImpl.redefineClasses(InstrumentationImpl.java:170) ~[?:1.8.0_40]
	at net.bytebuddy.dynamic.loading.ClassReloadingStrategy$Engine$1.apply(ClassReloadingStrategy.java:242) ~[byte-buddy-0.6.12.jar:?]
	at net.bytebuddy.dynamic.loading.ClassReloadingStrategy.load(ClassReloadingStrategy.java:154) ~[byte-buddy-0.6.12.jar:?]
	at net.bytebuddy.dynamic.DynamicType$Default$Unloaded.load(DynamicType.java:3486) ~[byte-buddy-0.6.12.jar:?]
```
Note that the method is not declared directly in the BlockDirt class, but rather it is declared in the Block class it extends from. Could that be why?

**Comment by raphw** [comment]
This is the reason. In order to intercept the method, Byte Buddy needs to add an overriding version to the redefined class. This is an unfortunate limitation in the JVM and is adressed by JEP-159: http://openjdk.java.net/jeps/159

This JEP is rather dormant but I hope it will be realized in the future.

**Comment by Anon10W1z** [comment]
That's unfortunate. Sorry to bother you. I'm rewriting the method in the superclass to check itself if it's an instance of the subclass as a workaround.

**Comment by raphw** [comment]
No, please ask. For longer questions, please also consider Stack Overflow with the `byte-buddy` tag or the mailing list.

**Comment by Anon10W1z** [comment]
Sounds good! Thanks again!

**Comment by raphw** [comment]
Should be fixed in 0.6.13 which was just released.
